src/components/audio/Audio.tsx

```diff
@@ -343,6 +344,41 @@ const Audio = ({
     }
   }, [elapsedTime, isPlaying])
 
+  // handle triggering of autoplay when current track ends
+  // (this is the flow when the next button is NOT clicked by the user)
+  // (if the next button is clicked by the user, the dapp client will handle autoplay logic)
+  const onNext = useCallback(() => {
+    // if autoplay is enabled and current song is close to end of queue,
+    // then trigger queueing of recommended tracks for autoplay
+    const isCloseToEndOfQueue = index + 2 >= queueLength
+    const isNotRepeating = repeatMode === RepeatMode.OFF
+    if (
+      webRef.current &&
+      queueAutoplay &&
+      !isShuffleOn &&
+      isNotRepeating &&
+      isCloseToEndOfQueue
+    ) {
+      postMessage(webRef.current, {
+        type: MessageType.QUEUE_AUTOPLAY,
```

Comment:
maybe we could make this action verb prefixed so it's clear what we want.. something like

FETCH_QUEUE_AUTOPLAY or REQUEST_QUEUE_AUTOPLAY or something like that